### PR TITLE
Feature/RENwidget

### PR DESCRIPTION
**Pull request by zackorenberg**
This new feature in the form of a Core Widget allows which allows the user to specify the remote enable level of any device, as well as a 'set default' for all devices, which is saved in the configuration file when changed.

### CODE REVIEW

File: LabDrivers/Tool.py
Comment (Bachibouzouk) [inline LabDrivers/Tool.py]: Use preferably docstrings, rather than a comment. A docstring will be displayed upon typing `help(set_ren)`. See https://www.python.org/dev/peps/pep-0257/ for best practices :)

File: LabDrivers/Tool.py
Comment (Bachibouzouk) [inline LabDrivers/Tool.py]: why not using `else if` as well?

File: LabDrivers/utils.py
Comment (Bachibouzouk) [inline LabDrivers/utils.py:757]: why don't you assign global variables in the higher scope of the module? (i.e., why inside a function?)

File: LabDrivers/utils.py
Comment (Bachibouzouk) [inline LabDrivers/utils.py:773]: I would add a reference to the manual where the ++loc command is described

File: LabGui.py
Comment (Bachibouzouk) [inline LabGui.py]: The philosophy we had with @ben741, was to write as little Widget code in `LabGui.py` as possible. These functions could be defined in the `RemoteEnableWidget.py` `add_widget_into_main` in the same way I did it here : 
https://github.com/LabGUI/LabGUI/blob/f34e0e73f5b9857f5fbbfb68cc4101a5c4b859f2/LabTools/CoreWidgets/LoadPlotWidget.py#L291

Comment (Bachibouzouk) [reply LabGui.py]: That way everything which has to do with the widget is located in a single file

File: LabTools/CoreWidgets/RemoteEnableWidget.py
Comment (Bachibouzouk) [inline LabTools/CoreWidgets/RemoteEnableWidget.py]: This comment is not very useful

File: LabTools/CoreWidgets/RemoteEnableWidget.py
Comment (Bachibouzouk) [inline LabTools/CoreWidgets/RemoteEnableWidget.py:311]: I would turn this into a docstring

File: LabTools/CoreWidgets/RemoteEnableWidget.py
Comment (Bachibouzouk) [inline LabTools/CoreWidgets/RemoteEnableWidget.py]: Commented lines do not belong to the code base ;)

File: LabTools/CoreWidgets/RemoteEnableWidget.py
Comment (Bachibouzouk) [inline LabTools/CoreWidgets/RemoteEnableWidget.py:286]: unsetting all what?

File: LabTools/CoreWidgets/RemoteEnableWidget.py
Comment (Bachibouzouk) [inline LabTools/CoreWidgets/RemoteEnableWidget.py:228]: leftover comments

File: LabTools/CoreWidgets/RemoteEnableWidget.py
Comment (Bachibouzouk) [inline LabTools/CoreWidgets/RemoteEnableWidget.py]:
```suggestion
        # aesthetics
```

File: LabTools/CoreWidgets/RemoteEnableWidget.py
Comment (Bachibouzouk) [inline LabTools/CoreWidgets/RemoteEnableWidget.py]: Maybe here you can describe a bit more in details and/or provide sources/manuals for interested readers/future users :)

**Review by Bachibouzouk — APPROVED** [review]
I would install flake8 (`pip install flake8`) and run it on the files you modified to spot PEP8 errors such as too many blank lines, missing whitespace around operator etc...

If you do it gradually, the code will be more and more pep8 compatible :)

File: LabDrivers/Tool.py
Comment (zackorenberg) [reply LabDrivers/Tool.py]: I did not use any else if's in the function, as there is overlap for REN levels. In this specific case there isn't, however, so I will implement that change before merging.

File: LabDrivers/utils.py
Comment (zackorenberg) [reply LabDrivers/utils.py:757]: I did not assign it as a global variable because it is only used in this function, and will never be subject to change.

File: LabTools/CoreWidgets/RemoteEnableWidget.py
Comment (zackorenberg) [reply LabTools/CoreWidgets/RemoteEnableWidget.py:286]: The way this widget works is that there can be a default REN enabled for all machines, which will automatically assert for each one on connection. When a user selects a level as the default, set_default(level) will automatically set the default REN value to the specified level for all devices, as well as graphically represent the change within the widget.

If the user wishes to remove a default REN, by passing level=None, it will remove the default (by setting it equal to None) and graphically represent the change within the widget by stating there is no default, by disabling the "Clear" button, and by updating each "line" such that the corresponding state has the option to be set as the default.

File: LabTools/CoreWidgets/RemoteEnableWidget.py
Comment (zackorenberg) [reply LabTools/CoreWidgets/RemoteEnableWidget.py]: lol, good catch :P

File: LabTools/CoreWidgets/RemoteEnableWidget.py
Comment (zackorenberg) [reply LabTools/CoreWidgets/RemoteEnableWidget.py]: I have updated it in the next commit. I could not find the website that precisely stated each level, but I copy-pasted it verbatim into the widget. (It took me a few hours to find and understand what exactly occurs for each level, piecing together information found in NI's limited documentation for their control_ren(mode) method, forum questions, and random user manuals for other 488.2 devices). Let me know if I should add anything else.

File: LabTools/CoreWidgets/RemoteEnableWidget.py
Comment (Bachibouzouk) [reply LabTools/CoreWidgets/RemoteEnableWidget.py:286]: This is a good description of what REN is useful for ! You could reuse it in the docstring of the widget :)

My comment was targeted to the sentence of the comment `# this will automatically update default_ren, as well as unsetting all`, did you mean `all level` then?

File: LabTools/CoreWidgets/RemoteEnableWidget.py
Comment (Bachibouzouk) [reply LabTools/CoreWidgets/RemoteEnableWidget.py]: > It took me a few hours to find and understand what exactly occurs for each level

Finding information and crunching it out in a more understandable format is very useful and can be very time consuming, thank you for taking this time :)!

Comment (Bachibouzouk) [reply LabTools/CoreWidgets/RemoteEnableWidget.py]: > I have updated it in the next commit. 

Did you push the commit? (FYI, if you copy the commit number in any github thread, it will automatically create a link to the commit ^^)

File: LabTools/CoreWidgets/RemoteEnableWidget.py
Comment (zackorenberg) [reply LabTools/CoreWidgets/RemoteEnableWidget.py]: removed

File: LabTools/CoreWidgets/RemoteEnableWidget.py
Comment (zackorenberg) [reply LabTools/CoreWidgets/RemoteEnableWidget.py:228]: removed

File: LabDrivers/utils.py
Comment (zackorenberg) [reply LabDrivers/utils.py:773]: I found a page for both ++loc and ++llo command and added it to the documentation

File: LabGui.py
Comment (zackorenberg) [reply LabGui.py]: corrected, and tested solely for intended functionality of the functions

**Comment by zackorenberg** [comment]
@Bachibouzouk, I have corrected the things we have discussed above, awaiting on your approval before merging

File: LabDrivers/Tool.py
Comment (Bachibouzouk) [inline LabDrivers/Tool.py:545]: I would direct you to https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings for the proper docstring formatting :), apart from that you can merge :)

**Review by Bachibouzouk — APPROVED** [review]
Only one comment about docstring :)

One think we could consistently make as well, would be to use `black` to auto-pep8 the code (this is a suggestion :), feel free to implement it or not)

**Pull request merged**
